Libraries/Connect/PackageInternal/ConnectError+GRPC.swift

```diff
@@ -25,6 +25,11 @@ extension ConnectError {
     ///                       passed in the headers block for gRPC-Web.
     ///
     /// - returns: A tuple containing the gRPC status code and an optional error.
+    @available(
+        swift,
+        deprecated: 100.0,
```

Comment:
This could be misleading. Is it dangerous to use 1.1 instead? If we end up needing a minor release before we're actually ready to use Swift 6 `package` visibility, could we just bump this out to 1.2? Would that cause havoc?

Reply:
This refers to the version of _Swift_ that this will be deprecated in - it's just an arbitrarily high number which doesn't show up in the warning message. Unfortunately, we cannot specify which version of the _package_ something will be deprecated in as part of this macro

Reply:
Oh. So why not set it to 6?

Reply:
I tested this out, but it triggers warnings if a consuming project upgrades their `Package.swift` to use Swift 6, which we don't want:

<img width="1624" alt="Screenshot 2024-10-19 at 8 42 01 AM" src="https://github.com/user-attachments/assets/b6a737a6-11fa-41d3-9126-b8ac7874a32f">